applications/fishing-map/src/features/map/map.slice.ts

```diff
@@ -1,34 +1,58 @@
 import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit'
 import uniqBy from 'lodash/uniqBy'
-import {
-  InteractionEvent,
-  ExtendedFeatureVessel,
-  ExtendedFeature,
-} from '@globalfishingwatch/react-hooks'
+import { InteractionEvent, ExtendedFeature } from '@globalfishingwatch/react-hooks'
 import GFWAPI from '@globalfishingwatch/api-client'
 import { resolveEndpoint } from '@globalfishingwatch/dataviews-client'
-import { DataviewDatasetConfig, Dataset, Vessel, DatasetTypes } from '@globalfishingwatch/api-types'
+import {
+  DataviewDatasetConfig,
+  Dataset,
+  Vessel,
+  DatasetTypes,
+  ApiEvent,
+} from '@globalfishingwatch/api-types'
 import { AsyncReducerStatus } from 'utils/async-slice'
 import { AppDispatch, RootState } from 'store'
 import {
   getRelatedDatasetByType,
+  selectEventsDataviews,
   selectTemporalgridDataviews,
 } from 'features/workspace/workspace.selectors'
 import { fetchDatasetByIdThunk, selectDatasetById } from 'features/datasets/datasets.slice'
 import { selectTimeRange } from 'features/app/app.selectors'
 
 export const MAX_TOOLTIP_VESSELS = 5
 
+export type ExtendedFeatureVessel = {
+  id: string
+  hours: number
+  dataset: Dataset
+  [key: string]: any
+}
+
+export type ExtendedFeatureEvent = ApiEvent & { dataset: Dataset }
+
+export type SliceExtendedFeature = ExtendedFeature & {
+  event?: ExtendedFeatureEvent
+  vessels?: ExtendedFeatureVessel[]
+}
+
+// Extends the default extendedEvent including event and vessels information from API
+export type SliceInteractionEvent = Omit<InteractionEvent, 'features'> & {
+  features: SliceExtendedFeature[]
```

Comment:
Comes from ExtendedFeature as the vessels list and events are app related instead of the hook responsability